packages/ui/src/Dashboard/DetailView/ActionsDetails/BuySell/BuySell.tsx

```diff
@@ -1,15 +1,143 @@
 import React, { Component } from 'react';
-import { createStyles, Grid, Theme, withStyles } from '@material-ui/core';
+import { Typography, Button, ButtonGroup, TextField, Box } from '@material-ui/core';
+import { createStyles, Grid, Theme, withStyles, WithStyles, Divider } from '@material-ui/core';
+import { AdditionalOrderType, Amount, OrderType, Quantity, Stock } from '@se/core';
+import { apiCall, getErrorMessage } from '../../../../utils/Util';
 
-const styles = (theme: Theme) => createStyles({});
+const styles = (theme: Theme) =>
+  createStyles({
+    root: {
+      display: 'flex',
+      flexDirection: 'column',
+      '& > *': {
+        margin: theme.spacing(1),
+      },
+    },
+  });
+
+interface Props extends WithStyles<typeof styles> {
+  stock: Stock;
+  orderType: OrderType;
+}
+
+interface State {
+  quantity: Quantity;
+  additionalOrderType: AdditionalOrderType;
+  price: Amount;
+}
+
+class BuySell extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      quantity: 0,
+      additionalOrderType: AdditionalOrderType.Market,
+      price: 0,
+    };
+  }
+
+  handleMarketOrder = () => this.setState({ additionalOrderType: AdditionalOrderType.Market });
+  handleLimitOrder = () => this.setState({ additionalOrderType: AdditionalOrderType.Limit });
+
+  handleBuySell = async () => {
+    const sendPrice = this.state.additionalOrderType == 'Limit' ? this.state.price : 0;
+    const payload = {
+      symbol: this.props.stock,
+      orderType: this.props.orderType,
+      additionalOrderType: this.state.additionalOrderType,
+      price: sendPrice,
+      quantity: this.state.quantity,
+    };
+    try {
+      const response = await apiCall('/v1/order/place', 'POST', payload);
+      const error = getErrorMessage(response);
+      if (error) {
+        console.log('Error');
```

Comment:
console.logs can be removed. For error case, we need to show an error on MessagePopup. so, a new prop for creating popup has to be passed from App.tsx to here.

Reply:
you can get showMessagePopup by extending AppProps like it's done in Landing.tsx